File: hw/ramos/u1/hw4MultiplicationTables/MultiplicationTables.py

while True:

    number = input("Enter a number to see its multiplication table (1-12): ")
    
    if not number.isdigit():
        print("Please enter a valid number.")
        continue
    
    number = int(number)
    
    if number < 1 or number > 12:
        print("You can only enter numbers from 1 to 12.")
        continue
    
    print(f"\nMultiplication table for {number}:")
    for i in range(1, 13):
        print(f"{number} x {i} = {number * i}")
    
    choice = input("\nDo you want to see another table? (yes/no): ").strip().lower()
    if choice != "yes":
        print("Goodbye!")
        break

File: hw/ramos/u1/hw4MultiplicationTables/MultiplicationTables.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function askNumber() {
    rl.question("Enter a number to see its multiplication table (1-12): ", (input) => {
        const number = parseInt(input);

        if (isNaN(number)) {
            console.log("Please enter a valid number.");
            askNumber();
            return;
        }

        if (number < 1 || number > 12) {
            console.log("You can only enter numbers from 1 to 12.");
            askNumber();
            return;
        }

        console.log(`\nMultiplication table for ${number}:`);
        for (let i = 1; i <= 12; i++) {
            console.log(`${number} x ${i} = ${number * i}`);
        }

        rl.question("\nDo you want to see another table? (yes/no): ", (choice) => {
            if (choice.trim().toLowerCase() === "yes") {
                askNumber();
            } else {
                console.log("Goodbye!");
                rl.close();
            }
        });
    });
}

askNumber();
